README.md

```diff
@@ -1,2 +1,29 @@
-# fastcode
+# fastcode  [![Build Status](https://travis-ci.com/matthaeusheer/fastcode.svg?branch=master)](https://travis-ci.com/matthaeusheer/fastcode)
+
 Group project for ETH course "How to Write Fast Numerical Code".
+
+## Testing
+
+This project uses [`criterion`](https://github.com/Snaipe/Criterion) for testing. Unit
+tests should be written straight in the `main.c` file of the algorithm. Integration tests
```

Comment:
I think it would be clearer to name the ```main.c``` of the algortihm (say pengu) after the algorithm itself. So the pengu algo directory would contain a file ```pengu.c``` with a corresponding ```pengu.h``` (and potential other stuff like utils or so) such that it is self explanatory. Importing main.h and having many main files floating around seems rather ugly. The files holding the source code shouldn't be intended to be run anyway.

Reply:
Mhhhh yeah sure I guess this would not really complicate the testing setup.

Regarding the fact that they are not supposed to run, that depends how we end up organising the unit tests, because if they are contained in the source files, you can use the files to run project specific unit tests.